koku/api/report/test/tests_queries.py

```diff
@@ -105,3 +114,137 @@ def test_group_data_by_list_missing_units(self):
                     ],
                 }
                 self.assertEqual(expected, out_data)
+
+
+def create_test_handler(mapper=None, params=None):
+    """Create a TestableReportQueryHandler using the supplied args."""
+    if not mapper:
+        mapper = {"filter": {}, "filters": {}}
+
+    if not params:
+        params = {"filter": {}, "group_by": {}, "order_by": {}}
+
+    class TestableReportQueryHandler(ReportQueryHandler):
+        """ A testable minimal implementation of ReportQueryHandler.
+
+             ReportQueryHandler can't be instantiated directly without first setting
+             a few attributes that are required by QueryHandler.__init__().
+         """
+
+        _mapper = Mock(
+            spec=ProviderMap,
+            _report_type_map=Mock(return_value=mapper, get=lambda x, y=None: mapper.get(x, y)),
+            _provider_map=Mock(return_value=mapper, get=lambda x, y=None: mapper.get(x, y)),
+        )
+
+    mock_params = Mock(
```

Comment:
There is an existing implementation that mocks out the QueryParams: https://github.com/project-koku/koku/blob/master/koku/api/iam/test/iam_test_case.py#L127 . And pattern for using the query handler here: https://github.com/project-koku/koku/blob/master/koku/api/report/test/aws/tests_queries.py#L67-L68. Might see if that can be reused.

Reply:
It can't be reused because `AWSQueryHandler.__init__()` implements logic that doesn't exist in `ReportQueryHandler`, primarily the setting of the `_mapper` attribute. That's why I had to create a testable sub-class to mock-out some of the logic that we've delegated to the provider-specific sub-classes.